### PR TITLE
added the auth key to the smtp driver options

### PR DESCRIPTION
**Pull request by kennymark**
## Proposed changes
By default the smtp driver options do not have the auth key available even tho it is essential for every smtp option. This just makes it easier for install to simply use smtp without having to depend on intellisense of their code editors to know the auth key exists on the smtp driver. 

## Types of changes

What types of changes does your code introduce?

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/adonisjs/mail/blob/master/CONTRIBUTING.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thetutlage** [comment]
Hey @kennymark Thanks for the PR.

It will be nice to also define these environment variables inside the [package.json file](https://github.com/adonisjs/mail/blob/develop/package.json#L131), so that they are added to the user `.env` file at the time of setup.

Also, I will rename them as `SMTP_USERNAME` and `SMTP_PASSWORD`

**Comment by kennymark** [comment]
> Hey @kennymark Thanks for the PR.
> 
> It will be nice to also define these environment variables inside the [package.json file](https://github.com/adonisjs/mail/blob/develop/package.json#L131), so that they are added to the user `.env` file at the time of setup.
> 
> Also, I will rename them as `SMTP_USERNAME` and `SMTP_PASSWORD`

I will do that if you dont mind later today

**Comment by thetutlage** [comment]
No at all. Take your time 😄

File: templates/config.txt [hunk 1]
```diff
@@ -46,6 +46,11 @@ const mailConfig: MailConfig = {
       driver: 'smtp',
       host: Env.get('SMTP_HOST') as string,
       port: Env.get('SMTP_PORT') as string,
+			auth: {
+				user: Env.get('SMTP_USERNAME') as string,
+				pass: Env.get('SMTP_PASSWORD') as string,
+				type: 'login',
+			}
```
Comment (RomainLanz) [inline templates/config.txt:53]: It seems that you have an issue with the formating of the code.

```suggestion
      auth: {
        user: Env.get('SMTP_USERNAME') as string,
        pass: Env.get('SMTP_PASSWORD') as string,
        type: 'login',
      }
```

**Pull request merged**